packages/devtools-map-bindings/src/tests/scopes.js

```diff
@@ -0,0 +1,159 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const { getLocationScopes } = require("../scopes");
+const { SourceMapConsumer, SourceMapGenerator } = require("source-map");
+
+jest.mock("devtools-utils/src/network-request");
+
+test("getLocationScopes", async () => {
+  const source = {
+    id: "source-mapped.js",
+    sourceMapURL: "source-mapped.js.map",
+    url: "http://example.com/source-mapped.js"
+  };
+
+  // Testing something like:
+  // `function () { var zero; { let one = 1; one++; } zero = 0; }`
+  var map = new SourceMapGenerator({
+    file: "source-mapped.js"
+  });
+
+  map.addMapping({
+    generated: {
+      line: 1,
+      column: 3
+    },
+    source: "foo.js",
+    original: {
+      line: 1,
+      column: 1
+    },
+    name: "zero"
+  });
+
+  map.addMapping({
+    generated: {
+      line: 11,
+      column: 15
+    },
+    source: "foo.js",
+    original: {
+      line: 33,
+      column: 5
+    },
+    name: "one"
+  });
+
+  map.addMapping({
+    generated: {
+      line: 11,
+      column: 35
+    },
+    source: "foo.js",
+    original: {
+      line: 33,
+      column: 9
+    },
+    name: "one"
+  });
+
+  map.addMapping({
+    generated: {
+      line: 20,
+      column: 5
+    },
+    source: "foo.js",
+    original: {
+      line: 50,
+      column: 1
+    },
+    name: "zero"
+  });
```

Comment:
we can clean this up after we add the parser code and can improve the test in one pass.